### PR TITLE
Moat/Anyio cherry pick

### PR DESCRIPTION
**Pull request by farmio**
I have cherry-picked some commits form #304

### CODE REVIEW

File: xknx/devices/climate.py [hunk 1]
```diff
@@ -58,13 +57,6 @@ def __init__(
         """Initialize Climate class."""
         # pylint: disable=too-many-arguments, too-many-locals, too-many-branches, too-many-statements
         super().__init__(xknx, name, device_updated_cb)
-        if isinstance(group_address_on_off, (str, int)):
```
Comment (marvin-w) [inline xknx/devices/climate.py:61]: Why is this removed?

**Review by marvin-w — CHANGES_REQUESTED** [review]
General idea looks fine. We just have to verify that all the devices still work. Also I think there is one code snippet that should not be deleted?

File: xknx/devices/climate.py (hunk 1 above)
Comment (farmio) [reply xknx/devices/climate.py:61]: now we can pass GroupAddress() objects to all group_address_* attributes. If something unparsable is passed, an error will be raised when instantiating the GroupAddress in RemoteValue.
There should be no need to double-check types here.

Comment (marvin-w) [reply xknx/devices/climate.py:61]: Ah, now it makes sense. We only use that in the constructor.

**Review by marvin-w — APPROVED** [review]
